apps/web/components/Routes.tsx

```diff
@@ -0,0 +1,33 @@
+import { useRouter } from "next/router";
+import { ReactElement, useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { selectUserState } from "../store/UserSlice";
+import { Loading } from "./Loading";
+
+interface RoutesProps {
+  children: ReactElement;
+}
+
+export default function Routes({ children }: RoutesProps): JSX.Element {
+  const userState = useSelector(selectUserState);
+  const router = useRouter();
+  const [loading, setLoading] = useState(true);
+
+  useEffect((): void => {
+    if (
```

Comment:
You do not check the `/auth/...` routes. This means that the OAuth2 and OTP routes cannot work. Maybe include some kind of regex to check all routes under `/auth`. Also, it would be easier if `/login` and `/register` were prefixed by `/auth`. I let you decide.

Reply:
Moved /login and /register under /auth